src/codegen/compile.ml

```diff
@@ -8207,283 +7983,74 @@ end (* MakeSerialization *)
 
 module Serialization = MakeSerialization(BumpStream)
```

Comment:
Unless we decide to have a single 32-64 bit compile.ml, I think we could even consider getting rid of the functor `MakeSerialiazation(Strm)`,  inlining its body here and just rename `BumpStream` to `Strm`.

Note: if we wanted to support forward compatibility from 64-bit to 32-bit via old skool Candid serialization then the 64-bit version could afford to use simple Blob serialization in main memory (we have headroom now) 
and then copy  the blob to stable memory, rather than streaming via BlobStream. 

I'm just not sure how you'd enable the forward compatible upgrade. Maybe with a hidden method (upgrade_via_candid) that sets a flag (yuck) before attempting the upgrade.